src/App.jsx

```diff
@@ -14,45 +16,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const rev = 'reverse';
+export const alphabet = 'alphabet';
+export const alphabetRev = 'alphabet reverse';
+export const length = 'length';
+export const lengthRev = 'length reverse';
+
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [sortField, setSortField] = useState('');
+
+  const sortAlphabet = () => {
+    if (sortField.includes(rev)) {
+      setVisibleGoods([...visibleGoods].sort((a, b) => b.localeCompare(a)));
+      setSortField(alphabetRev);
+    } else {
+      setVisibleGoods([...visibleGoods].sort((a, b) => a.localeCompare(b)));
+      setSortField(alphabet);
+    }
+  };
+
+  const sortLength = () => {
+    if (sortField.includes(rev)) {
+      setVisibleGoods([...visibleGoods].sort((a, b) => b.length - a.length));
+      setSortField(lengthRev);
+    } else {
+      setVisibleGoods([...visibleGoods].sort((a, b) => a.length - b.length));
+      setSortField(length);
+    }
+  };
+
+  const reverse = () => {
+    setVisibleGoods([...visibleGoods].reverse());
+    if (sortField.includes(rev)) {
+      setSortField(sortField.split(' ')[0]);
+    } else {
+      setSortField(`${sortField} ${rev}`);
+    }
+  };
```

Comment:
![image](https://github.com/mate-academy/react_list-of-goods-js/assets/96265844/0848469b-5774-4b04-bd0d-30caf411c1ff)